src/main/java/com/Acrobot/ChestShop/Listeners/Economy/Plugins/ReserveListener.java

```diff
@@ -0,0 +1,176 @@
+
+        if (lastSeen != null && provided()) {
+            event.setAmount(economyAPI.getHoldings(event.getAccount(), event.getWorld().getName()));
+        }
+    }
+
+    @EventHandler
+    public void onCurrencyCheck(CurrencyCheckEvent event) {
+        if (event.hasEnough()) {
+            return;
+        }
+        final OfflinePlayer lastSeen = Bukkit.getOfflinePlayer(event.getAccount());
+
+        if (lastSeen != null && provided()) {
+            event.hasEnough(economyAPI.hasHoldings(event.getAccount(),
+                                                                                         event.getAmount(),
+                                                                                         event.getWorld().getName()));
+        }
+    }
+
+    @EventHandler
+    public void onAccountCheck(AccountCheckEvent event) {
+        if (event.hasAccount()) {
+            return;
+        }
+        final OfflinePlayer lastSeen = Bukkit.getOfflinePlayer(event.getAccount());
+        event.hasAccount(lastSeen != null && provided() && economyAPI.hasAccount(event.getAccount()));
+    }
+
+    @EventHandler
+    public void onCurrencyFormat(CurrencyFormatEvent event) {
+        if (!event.getFormattedAmount().isEmpty()) {
+            return;
+        }
+
+        if (provided()) {
+            event.setFormattedAmount(economyAPI.format(event.getAmount()));
+        }
+    }
+
+    @EventHandler
+    public void onCurrencyAdd(CurrencyAddEvent event) {
+        if (event.isAdded()) {
+            return;
+        }
+        final OfflinePlayer lastSeen = Bukkit.getOfflinePlayer(event.getTarget());
+
+        if (lastSeen != null && provided()) {
+            economyAPI.addHoldings(event.getTarget(), event.getAmount(), event.getWorld().getName());
+        }
+    }
+
+    @EventHandler
+    public void onCurrencySubtraction(CurrencySubtractEvent event) {
+        if (event.isSubtracted()) {
+            return;
+        }
+        final OfflinePlayer lastSeen = Bukkit.getOfflinePlayer(event.getTarget());
+
+        if (lastSeen != null && provided()) {
+            economyAPI.removeHoldings(event.getTarget(), event.getAmount(), event.getWorld().getName());
+        }
+    }
+
+    //Copied from VaultListener as Reserve doesn't have a dedicated transfer api until 1.0.11
+    @EventHandler
+    public static void onCurrencyTransfer(CurrencyTransferEvent event) {
+        if (event.hasBeenTransferred()) {
+            return;
+        }
+
+        CurrencySubtractEvent currencySubtractEvent = new CurrencySubtractEvent(event.getAmount(), event.getSender(), event.getWorld());
+        ChestShop.callEvent(currencySubtractEvent);
+
+        if (!currencySubtractEvent.isSubtracted()) {
+            return;
+        }
+
+        CurrencyAddEvent currencyAddEvent = new CurrencyAddEvent(currencySubtractEvent.getAmount(), event.getReceiver(), event.getWorld());
+        ChestShop.callEvent(currencyAddEvent);
+    }
```

Comment:
This is not a good way of handling the CurrencyTransferEvent and having this in the VaultListener is equally bad: It would result in the subtract and adding happening twice if both listeners react on it. Either mark it as transferred after the events are completed or use the economy API to process it.

Besides the fact that this event isn't even called anywhere and that I'm questioning the usefulness and backwards compatibility of converting it - we would basically have to call the subtract, transfer and add events to make sure that other plugins (and even internal stuff like taxes) properly react on it. Also they use non-transaction-style economy actions (like removing or adding money without a source/target) so I'm not sure that using this transaction style suits the plugin at all.

Obviously the discussion about the transactions isn't really your concern, I just want to show my reasoning behind why using this transfer reaction twice would lead to issues further down the road.

Reply:
Wasn't sure, I was just using what was already made in the example of the VaultListener.

Reply:
Either way, I've created a method that will be in 1.0.11 of Reserve if this functionality is needed. https://github.com/TheNewEconomy/Reserve/blob/chat/src/net/tnemc/core/economy/EconomyAPI.java#L663

Reply:
I just replaced this with calls to the other listener methods for now. If we end up using that event in ChestShop in the future and you think it would be better to use your native methods for transfers feel free to PR that in.